src/Microsoft.Framework.Configuration.FileExtensions/FileConfigurationExtensions.cs

```diff
@@ -49,7 +49,13 @@ public static string GetBasePath(this IConfigurationBuilder configurationBuilder
                 return (string)basePath;
             }
 
-            return string.Empty;
+#if DOTNET
+            return AppContext.BaseDirectory;
+#else
+            var baseDir = AppDomain.CurrentDomain.GetData("APP_CONTEXT_BASE_DIRECTORY");
+
+            return (baseDir != null) ? (string)baseDir : AppDomain.CurrentDomain.BaseDirectory;
```

Comment:
Actually, should we append `?? string.Emtpty` to both returns to make sure null is never returned (so that `Path.Combine()` is happy?